app/master/cluster_master.py

```diff
@@ -62,12 +62,34 @@ def api_representation(self):
             'slaves': slaves_representation,
         }
 
-    def builds(self):
+    def builds(self, offset: int=None, limit: int=None) -> List['Build']:
         """
-        Returns a list of all builds
-        :rtype: list[Build]
+        Returns a list of all builds.
+        :param offset: The starting index of the requested build
+        :param limit: The number of builds requested
         """
-        return self._all_builds_by_id.values()
+        num_builds = len(self._all_builds_by_id)
+        highest_amount = min(num_builds, Configuration['pagination_max_limit'])
+
+        offset = offset if offset is not None else Configuration['pagination_offset']
+        limit = limit if limit is not None else Configuration['pagination_limit']
+
+        # Remove any negative values
+        offset = max(offset, 0)
+        limit = max(limit, 0)
+
+        # If limit is set higher than the number of builds/max limit, reduce limit
+        limit = highest_amount if limit > highest_amount else limit
+
+        # Requested offset too high should yield no results
+        if offset > num_builds:
+            return []
+
+        first_build_id = offset + 1
```

Comment:
Ah, so it's because build id's are 1 indexed and offset is 0 indexed. I feel like that might be confusing. Personally if I queried for `offset=10,limit=1` i'd expect to get build_id 10 back. I feel like it might make sense to either start 0 indexing the builds or 1 index the offset. If we 1 indexed the offset, maybe it would make more sense to name it `start`? (I think that might make more sense anyway).

I can't decide if the current behavior is actually confusing, or if i'm just looking for something to complain about. Opinion @josephharrington ?